src/lib/storage/session.js

```diff
@@ -235,6 +238,31 @@ SessionStorage.prototype = {
   },
 
   /**
+   * Get the selected tab index for a group.
+   *
+   * @param {ChromeWindow} chromeWindow
+   * @param {Number} groupID - the groupID
+   */
+  getGroupSelectedIndex: function(chromeWindow, groupID) {
+    let groupsData = this._getGroupsData(chromeWindow);
+    let currentGroup = groupsData[groupID];
+    return currentGroup == null ? 0 : currentGroup.selectedIndex;
+  },
+
+  /**
+   * Set the selected tab index for a group.
+   *
+   * @param {ChromeWindow} chromeWindow
+   * @param {Number} groupID - the groupID
+   * @param {Number} index - the new selected index
+   */
+  setGroupSelectedIndex: function(chromeWindow, groupID, index) {
+    let groupsData = this._getGroupsData(chromeWindow);
+    groupsData[groupID].selectedIndex = index;
```

Comment:
`renameGroup` can only be called if there is information about a group since it exists. The issue here is that you select the group whenever a tab is changed/added, so if there are no groups and the user adds a tab, an exception is thrown here.

`SessionStorage.getGroups` ensures that there is at least one group, so this is not an issue for `renameGroup`. I cannot think of an edge case where this might get triggered without having a group loaded, so I would not add a check there.